include/net/wifi_ready.h

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+
+#ifndef WIFI_READY_H__
+#define WIFI_READY_H__
+
+#define WPA_SUPP_EVENTS (NET_EVENT_WPA_SUPP_READY) | \
+			(NET_EVENT_WPA_SUPP_NOT_READY)
+
+typedef void (*start_app_callback_t)();
+
+typedef struct {
+	start_app_callback_t is_wifi_ready;
+} app_callback_t;
+
+void register_supp_events(app_callback_t cb);
```

Comment:
This library takes care of both WPA supplicant and NET_IF events, so, this should be `register_wifi_ready_callback`

Reply:
also, the callbacks should be per-iface, in fact this entire library should be per-iface even though we might not implement it fully in this PR, cannot keep changing public interface.